Tools/TestWebKitAPI/Tests/WebKitGLib/TestDownloads.cpp

```diff
@@ -827,6 +827,31 @@ static void testDownloadEphemeralContext(Test* test, gconstpointer)
     g_file_delete(destFile.get(), nullptr, nullptr);
 }
 
+
+static void testDownloadCancelledByEmptyDestination(Test* test, gconstpointer)
+{
+    GRefPtr<WebKitWebsiteDataManager> manager = adoptGRef(webkit_website_data_manager_new_ephemeral());
+    test->assertObjectIsDeletedWhenTestFinishes(G_OBJECT(manager.get()));
+    GRefPtr<WebKitWebContext> context = adoptGRef(webkit_web_context_new_with_website_data_manager(manager.get()));
+    test->assertObjectIsDeletedWhenTestFinishes(G_OBJECT(context.get()));
+    g_assert_true(webkit_web_context_is_ephemeral(context.get()));
+
+    GRefPtr<GMainLoop> mainLoop = adoptGRef(g_main_loop_new(nullptr, TRUE));
+    GRefPtr<WebKitDownload> download = adoptGRef(webkit_web_context_download_uri(context.get(), kServer->getURIForPath("/test.pdf").data()));
+    test->assertObjectIsDeletedWhenTestFinishes(G_OBJECT(download.get()));
+    g_signal_connect(download.get(), "decide-destination", G_CALLBACK(+[](WebKitDownload* download, const gchar* suggestedFilename, gpointer) {
+        webkit_download_set_destination(download, "");
```

Comment:
> Eh, is this really expected to work? The API does not say that passing an empty string is allowed. I would expect a critical.

Dunno but the implementation does [handle that](https://github.com/WebKit/WebKit/blob/fcd1d36de2ca84f2d8736165fbc08cd6251f9578/Source/WebKit/NetworkProcess/NetworkProcess.cpp#L2042-L2043) explicitly.

> Also: you need to return TRUE here, otherwise it's undefined behavior.

Fixed. I copied the code from the test above.